### ISSUE TITLE
broken with new httparty 0.14

### ISSUE THREAD

**Issue by benhutton** [soundcloud/soundcloud-ruby#53]
This line https://github.com/soundcloud/soundcloud-ruby/blob/master/lib/soundcloud/client.rb#L154

Is affected by this change: https://github.com/jnunemaker/httparty/commit/c4a031d70d09a5d1dff7661ad71030bddbcf6e71

The result is that you end up just getting back an `HTTParty::Response` object that will not respond to the methods you expect it to: https://github.com/soundcloud/soundcloud-ruby/blob/master/lib/soundcloud/client.rb#L159

In a related httparty issue (https://github.com/jnunemaker/httparty/issues/489), the author says that the new behavior is correct. Thus gems that were leveraging the old behavior are going to have to update.

**Comment by ryanckulp** [comment]
Looking forward to a fix here.

**Comment by moritzgloeckl** [comment]
I tried reverting to `httparty` version `0.3.16` but that doesn't help. Any other ideas?

**Comment by benhutton** [comment]
our hack has been to wrap the `parsed_response` in an `OpenStruct`. So something like `OpenStruct.new(response.parsed_response)`

**Comment by moritzgloeckl** [comment]
Good idea, I just forked the project and added a fix as suggested [here](https://github.com/jnunemaker/httparty/issues/489). I didn't test it much but it seems to be working for now. Link: [Soundcloud-Ruby-Fix](https://github.com/moritzgloeckl/soundcloud-ruby)

**Comment by ryanckulp** [comment]
@moritzgloeckl thanks! i'll give it a shot.

**Comment by sagarbommidi** [comment]
Any updates on this pls ?
I also getting the same error when i am using rails5.

**Comment by noahfeder** [comment]
I found a workaround for this (and other similar API-wrapping gems).
Instead of using the parsed_response method, try something like this:
```ruby
client = Soundcloud.new(:client_id => ENV['SOUNDCLOUD_CLIENT_ID'])
response = client.get('/tracks', :limit => 1)
parsed_response = JSON.load(JSON.generate(response))
```

Generating and then loading/parsing the HTTParty response cleans up whatever mess is throwing a lot of people off.

**Comment by moritzgloeckl** [comment]
@noahfeder  Well but that's just a temporary fix, actually this should be fixed in the library and not by the programmer using the library... You could try my version, which works fine for me!

**Comment by noahfeder** [comment]
@moritzgloeckl will definitely take a look!

**Comment by grobie** [comment]
I pushed a fix to a branch right now. Will release a new patch version soon.

**Comment by grobie** [comment]
Fixed in v0.3.3. I'll push the gem to rubygems.org as soon as I got the credentials. Feel free to build your own gem in the meantime.
